src/lib/components/toaster/toasts.ts

```diff
@@ -64,3 +64,25 @@ export function postponeDismissal(id: number, timeout: number): void {
 export function dismissToast(id: number): void {
   toasts.update((all) => all.filter((t) => t.id !== id));
 }
+
+export type doToastOpts<T> = Omit<Omit<Toast, 'id'>, 'content'> & {
+  success: (_: T) => string;
+  failure: (err: unknown) => string;
+};
+
+export async function toastDo<T>(
+  f: () => T | Promise<T>,
+  opts: doToastOpts<T> = {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    success: (_: T) => 'Success!',
+    failure: (err: unknown) => `An error occured while performing action: ${err}`,
+  }
+) {
+  try {
```

Comment:
If await is called on this function (e.g. `await toastDo(/*...*/);`), then it will block on this try/catch block. I do not think this currently causes any problems, but it is probably not intentional to block the parent scope on the toast function (future contributors may see "async" and want to await it, but they really only should when they want to block on the toast function). A non-blocking approach would call for then/catch over try/catch and return without waiting for the resolution.

Reply:
This doesn't matter; the user does `await toastDo` if they wish to wait for the given callback. `toast` itself is not async, so this has virtually no impact.